### PR TITLE
Reduced focal loss implementation is incorrect for binary case

### PR DESCRIPTION
**Pull request by MrPrajwalB**
The current implementation is same as the one given in original [paper ](https://arxiv.org/abs/1903.01347). Although, it doesn't work as intended when value of reduced threshold is not 0.5. As per the paper, the focal term should be 1 when pt is less than reduced threshold and then exponentially decrease from 1 to 0 as pt goes from reduced threshold to 1. By this logic, the value of focal term should be 1 at pt equal to reduced threshold from both sides of the curve. From RHS the value is given as (1-pt/thresh) to power gamma multiplied by log pt. This will only be 1 at pt=thresh, when thresh=0.5, otherwise not. 

So the correct formula focal term for pt>thresh should be  ((1-pt)/(1-thresh) to the power gamma multiplied by log pt.

The corrected implementation goes hand in hand with the one done for  softmax_focal_loss_with_logits which works as intended in the paper.

### CODE REVIEW

**Comment by BloodAxe** [comment]
Hmmm, indeed!

Thanks for spotting this bug. I've never get any decent results with reduced focal loss, and this may explain why :)

focal_term
